content/posts/2021/reana-client-python-api.md

```diff
@@ -0,0 +1,170 @@
+---
+title: "How to run workflows using reana-client Python API"
+date: 2021-02-22T15:00:00+01:00
+tags:
+  - API
+  - client
+  - Python
+---
+
+Here you can find a step by step guide on how to submit workflows to
+REANA using the `reana-client` Python API. This is a simple example,
+for illustrative purposes, using the serial workflow engine and a
+simple [ROOT6 / RooFit analysis](https://github.com/reanahub/reana-demo-root6-roofit/) example.
+We will create a new workflow, run it and retrieve its results using
+the `reana-client` Python API.
+<!--more-->
+
+Install `reana-client`:
+```
+$ pip install reana-client==0.7.0
+```
+
+Folowing instructions can be executeed inside a Python interpreter,
+or a Python file.
+
+First we need to provide the minimum configuration:
+- Point the REANA service URL:
+```python
+# with `reana-client` it would be
+#   `export REANA_SERVER_URL=https://reana.cern.ch`
+import os
+
+os.environ['REANA_SERVER_URL'] = 'https://reana.cern.ch'
+```
+
+- Provide our REANA access token:
+```python
+# with `reana-client` it would be
+#  `export REANA_ACCESS_TOKEN=XXXXXXX`
+from getpass import getpass
+
+my_reana_token = \
+  os.getenv("REANA_ACCESS_TOKEN") or getpass('Enter your REANA token: ')
+```
+
+Now we will choose a name for the workflow and the kind of
+workflow specification we will use:
+```python
+my_workflow_name = 'root6-roofit'
+workflow_type = 'serial'
+```
+
+Let us specify inputs and workflows specification:
+- So first we will provide it's inputs: (files and parameters)
+  (see the [reana.yaml](https://github.com/reanahub/reana-demo-root6-roofit/blob/f29e98b482fe8cb801735ac2fa48bc01e6cc05b7/reana.yaml#L2-L9)
+  code that corresponds to it)
+```python
+my_inputs = {
+    'files': [
+        'code/gendata.C',
+        'code/fitdata.C'
+    ],  # A list of files your analysis will be using
+    'parameters': {
+        'events': '20000',
+        'data': 'results/data.root',
+        'plot': 'results/plot.png',
+    }  # Parameters for your workflow
+}
+```
+
+- Then we will provide the actual workflow specification,
+  in this case using serial workflow:
+  (see the [reana.yaml](https://github.com/reanahub/reana-demo-root6-roofit/blob/f29e98b482fe8cb801735ac2fa48bc01e6cc05b7/reana.yaml#L13-L22)
+  code that corresponds to it
+  )
+```python
+workflow_json = {
```

Comment:
Name it `my_workflow`, by analogy with `my_inputs`.  The fact that is comes from JSON is not significant.  (And one could ask why we don't call `my_inputs_json` in the former case).  Better to just use my_code, my_inputs, my_workflow, my_results, etc.